### PR TITLE
Refactored saving of bank holiday dates to make sure they are cached …

### PR DESCRIPTION
**Pull request by LAA-NeilH**
## What

[Link to story](https://dsdmoj.atlassian.net/jira/software/c/projects/LEP/boards/1143?modal=detail&selectedIssue=LEP-197)

Describe what you did and why.
Refactored saving of bank holiday dates to make sure they are cached after 10 days.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by starswan** [comment]
It would be nice to write the (missing) test so that we could refactor this safely into a faraday cache at some point

**Comment by LAA-NeilH** [comment]
@starswan Will add the missing test

File: app/models/bank_holiday.rb [hunk 1]
```diff
@@ -18,5 +18,6 @@ def self.populate_dates
     rec = BankHoliday.first || BankHoliday.new
     rec.dates = GovukBankHolidayRetriever.dates
     rec.save!
+    rec.touch
```
[the remark this answers is not in the document]
Comment (LAA-NeilH) [reply app/models/bank_holiday.rb:21]: I originally did that Stephen suggested otherwise so I went with that.

**Comment by davidread** [comment]
LEP-203 means we don't need a better test in this PR

**Review by davidread — APPROVED** [review]

**Pull request merged**